cpp/src/neighbors/ivf_pq/detail/ivf_pq_search_half_int64_t.cu

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * NOTE: this file is generated by generate_ivf_pq.py
+ *
+ * Make changes there and run in this directory:
+ *
+ * > python generate_ivf_pq.py
+ *
+ */
+
+#include <cuvs/neighbors/ivf_pq.hpp>
+
+#include "../ivf_pq_search.cuh"
+
+namespace cuvs::neighbors::ivf_pq {
+
+#define CUVS_INST_IVF_PQ_SEARCH(T, IdxT)                                        \
+  void search(raft::resources const& handle,                                    \
+              const cuvs::neighbors::ivf_pq::search_params& params,             \
+              cuvs::neighbors::ivf_pq::index<IdxT>& index,                      \
+              raft::device_matrix_view<const T, IdxT, raft::row_major> queries, \
+              raft::device_matrix_view<IdxT, IdxT, raft::row_major> neighbors,  \
+              raft::device_matrix_view<float, IdxT, raft::row_major> distances) \
+  {                                                                             \
+    cuvs::neighbors::ivf_pq::detail::search(                                   \
+      handle, params, index, queries, neighbors, distances);  \
+  }
```

Comment:
I know it's a bit out of the scope of this PR, but I think it would make sense to remove the `#define CUVS_INST_IVF_PQ_SEARCH` in the generated code and put it in a separate header. Like we do with `CUVS_INST_IVF_PQ_BUILD_EXTEND` and a few others: for consistency and to reduce code duplication.

Reply:
I might prefer to remove the macro. Created #269 to track this.